cmd/builder/internal/builder/main.go

```diff
@@ -79,6 +80,9 @@ func Generate(cfg Config) error {
 		}
 		cfg.Logger.Info("You're building a distribution with non-aligned version of the builder. Compilation may fail due to API changes. Please upgrade your builder or API", zap.String("builder-version", defaultOtelColVersion))
 	}
+
+	// confmap factories were introduced in v0.99.0.
+	cfg.SupportsConfmapFactories = semver.Compare("v"+cfg.Distribution.OtelColVersion, "v0.99.0") > -1
```

Comment:
For the `RequireOtelColVersion` we do this check on cmd/builder/internal/builder/config.go on a method from config. Should we do the same here?

Reply:
Great catch! I missed this, but it would be nice to consolidate where we handle backwards compatibility. I've refactored this so all checks are done in the same place.